src/osd/ReplicatedBackend.cc

```diff
@@ -663,6 +771,186 @@ void ReplicatedBackend::op_commit(
   }
 }
 
+void ReplicatedBackend::op_commit_no_pg_lock(
+  InProgressOp *op)
+{
+
+  dout(20) << __func__ << " " << __LINE__ << " enter tid:" << op->tid << dendl;
+  op->op->comp_lock.Lock();
+
+  dout(10) << __func__ << ": " << op->tid << dendl;
+  if (op->op)
+  {
+    op->op->mark_event("op_commit");
+  }
+
+  op->waiting_for_commit.erase(op->shard);
+  if (op->waiting_for_commit.empty()) {
+
+    if (op->on_commit) {
+      if (op->waiting_for_applied.empty() && !op->op->completed) {
+	dout(20) << __func__ << " " << __LINE__ << 
+		" call on_commit->complete(2): " << op->tid << dendl;
+	// send ack and add completion q
+	op->on_commit->complete(OP_COMP_ACK_COMP);
+	op->op->completed = true;
+      } else {
+	// send ack
+	dout(20) << __func__ << " " << __LINE__ << 
+		" call on_commit->complete(1): " << op->tid << dendl;
+	op->on_commit->complete(OP_COMP_ACK);
+      }
+    } else {
+      assert(0);
+    } 
+    op->op->all_committed = true;
+    op->on_commit = 0;
+  }
+
+  if (op->done()) {
+    //op->unlock();
+    op->op->comp_lock.Unlock();
+    in_progress_ops_lock.Lock();
+    if (in_progress_ops.count(op->tid)) {
+      dout(20) << __func__ << " " << __LINE__ << 
+	      " op deleted out tid:" << op->tid << dendl;
+      in_progress_ops.erase(op->tid);
+    }
+    in_progress_ops_lock.Unlock();
+    return;
+  }
+  op->op->comp_lock.Unlock();
+  dout(20) << __func__ << " " << __LINE__ << " out tid:" << op->tid << dendl;
+}
+
+void ReplicatedBackend::sub_op_modify_reply_no_pg_lock(OpRequestRef op)
```

Comment:
@athanatos I think old version codes which based on pg lock mean sub_op_modify_rep(). old version codes can be removed if op lock completion is finished.